core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -54,7 +54,10 @@ private[spark] abstract class Task[T](
     val partitionId: Int,
     // The default value is only used in tests.
     val metrics: TaskMetrics = TaskMetrics.registered,
-    @transient var localProperties: Properties = new Properties) extends Serializable {
+    @transient var localProperties: Properties = new Properties,
+    val jobId: Option[Int] = None,
```

Comment:
so Task is a private spark api so there is no issues breaking api.  You don't need to deprecate and add new, you can just change and fix up everywhere that calls it or extends it.
There are a handful of test classes that extend it that would need to be fixed up.  Let me think about it a bit more.